src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -89,7 +122,32 @@ public Query(string queryText, ConnectionInfo connection, QueryExecutionSettings
                             batch.EndLocation.LineNumber - 1,
                             batch.EndLocation.ColumnNumber - 1),
                         index, outputFactory));
+                        
             Batches = batchSelection.ToArray();
+
+            // Create our batch lists
+            BeforeBatches = new List<Batch>();
+            AfterBatches = new List<Batch>();
+
+            // Checking settings for execution plan options 
+            if (querySettings.ExecutionPlanOptions.IncludeEstimatedExecutionPlanXml) 
+            {
+                // Check support level 
+                if (this.doesSupportExecutionPlan(connection))
+                {
+                    // Enable set showplan xml
+                    addBatch(string.Format(s_SetShowPlanXml, s_On), BeforeBatches, streamOutputFactory);
```

Comment:
yes, adding a batch before and after based on an option

---
In reply to: [95651178](https://github.com/Microsoft/sqltoolsservice/pull/213#discussion_r95651178) [](ancestors = 95651178)